### PR TITLE
Correctly deduplicate hunk locks

### PR DESCRIPTION
**Pull request by estib-vega**
Deduplicate hunk text by the property values they contain

### CODE REVIEW

File: apps/desktop/src/lib/utils/filters.ts [hunk 1]
```diff
@@ -2,3 +2,19 @@
...
+	const propertyKeys = Object.keys(value) as (keyof T)[];
+	if (propertyKeys.length > 0) {
+		return array.findIndex((v) => propertyKeys.every((key) => v?.[key] === value[key])) === index;
+	}
+
+	const objectPropertyKeys = Object.getOwnPropertyNames(value) as (keyof T)[];
+	return (
+		array.findIndex((v) => objectPropertyKeys.every((key) => v?.[key] === value[key])) === index
+	);
+}
```
Comment (mtsgrd) [inline apps/desktop/src/lib/utils/filters.ts]: For the purpose of keeping things simple, it feels like we could live without this section for non-enumerable properties? https://stackoverflow.com/a/22658584

Also, perhaps deepUnique would be a fitting name?

Comment (estib-vega) [reply apps/desktop/src/lib/utils/filters.ts]: We can 👍. 

Regarding the naming, I think 'deep' would imply that we compare the values recursively, while we only check one level inside. 

Maybe something like `uniqueByObjectValue`?

File: apps/desktop/src/lib/utils/filters.test.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	});
+});
+
+describe.concurrent('uniqeByPropValues', () => {
+	class Test {
+		constructor(
+			public id: number,
+			public name: string
+		) {}
+		doSomwthing() {
```
Comment (ndom91) [inline apps/desktop/src/lib/utils/filters.test.ts]: Typo? :joy:

**Review by ndom91 — APPROVED** [review]
LGTM. I agree with @estib-vega about the naming thing though, `deep` for me would imply we're comparing them recursively

File: apps/desktop/src/lib/utils/filters.test.ts (hunk 2 above)
Comment (estib-vega) [reply apps/desktop/src/lib/utils/filters.test.ts]: 😬

**Comment by mtsgrd** [comment]
Yes, I agree. I thought I was using my brain when clearly I wasn't!

**Pull request merged**